engine/verification/assigner/engine_test.go

```diff
@@ -1 +1,435 @@
+		state:            &protocol.State{},
+		snapshot:         &protocol.Snapshot{},
+		metrics:          &module.VerificationMetrics{},
+		tracer:           trace.NewNoopTracer(),
+		assigner:         &module.ChunkAssigner{},
+		chunksQueue:      &storage.ChunksQueue{},
+		newChunkListener: &module.NewJobListener{},
+		verIdentity:      unittest.IdentityFixture(unittest.WithRole(flow.RoleVerification)),
+		notifier:         &mockassigner.ProcessingNotifier{},
+		indexer:          &storage.Indexer{},
+	}
+
+	for _, apply := range options {
+		apply(s)
+	}
+	return s
+}
+
+// createContainerBlock creates and returns a block that contains an execution receipt, with its corresponding chunks assignment based
+// on the input options.
+func createContainerBlock(options ...func(result *flow.ExecutionResult, assignments *chunks.Assignment)) (*flow.Block, *chunks.Assignment) {
+	result, assignment := test.CreateExecutionResult(unittest.IdentifierFixture(), options...)
+	receipt := &flow.ExecutionReceipt{
+		ExecutorID:      unittest.IdentifierFixture(),
+		ExecutionResult: *result,
+	}
+	// container block
+	header := unittest.BlockHeaderFixture()
+	block := &flow.Block{
+		Header: &header,
+		Payload: &flow.Payload{
+			Receipts: []*flow.ExecutionReceipt{receipt},
+		},
+	}
+	return block, assignment
+}
+
+// NewAssignerEngine returns an assigner engine for testing.
+func NewAssignerEngine(s *AssignerEngineTestSuite) *Engine {
+
+	e := New(zerolog.Logger{},
+		s.metrics,
+		s.tracer,
+		s.me,
+		s.state,
+		s.assigner,
+		s.chunksQueue,
+		s.newChunkListener,
+		s.indexer)
+
+	e.withBlockConsumerNotifier(s.notifier)
+
+	// mocks identity of the verification node
+	s.me.On("NodeID").Return(s.verIdentity.NodeID)
+
+	return e
+}
+
+// TestNewBlock_HappyPath evaluates that passing a new finalized block to assigner engine that contains
+// a receipt with one assigned chunk, results in the assigner engine passing the assigned chunk to the
+// chunks queue and notifying the job listener of the assigned chunks.
+func TestNewBlock_HappyPath(t *testing.T) {
+	t.Parallel()
+
+	s := SetupTest()
+	e := NewAssignerEngine(s)
+
+	// creates a container block, with a single receipt, that contains
+	// one assigned chunk to verification node.
+	containerBlock, assignment := createContainerBlock(
+		test.WithChunks(
+			test.WithAssignee(s.myID())))
+	result := &containerBlock.Payload.Receipts[0].ExecutionResult
+	s.mockStateAtBlockID(result.BlockID)
+	chunksNum := s.mockChunkAssigner(result, assignment)
+	require.Equal(t, chunksNum, 1) // one chunk should be assigned
+
+	// mocks processing assigned chunks
+	// each assigned chunk should be stored in the chunks queue and new chunk listener should be
+	// invoked for it.
+	// Also, once all receipts of the block processed, engine should notify the block consumer once, that
+	// it is done with processing this chunk.
+	s.chunksQueue.On("StoreChunkLocator", mock.Anything).Return(true, nil).Times(chunksNum)
```

Comment:
the mocked chunks queue could only verify the number of times it's called. 

If we use the real one, we just need to make a simple consumer from the chunks queue that reads all the stored chunks, and verify if they are correct.

Reply:
This is also going to addressed in https://github.com/dapperlabs/flow-go/issues/5294.